### PR TITLE
[test] Modernize NAMD and GridTools tests

### PR DESCRIPTION
**Pull request by ChristopherBignamini**
Updated tests: 
 - NAMD
 - GridTools

### CODE REVIEW

File: cscs-checks/apps/namd/namd_check.py [hunk 1]
```diff
@@ -42,6 +48,21 @@ def __init__(self, arch, scale, variant):
                 self.num_tasks = 16
                 self.num_tasks_per_node = 1
 
+    @rfm.run_after('init')
+    def set_valid_prgenv(self):
+        if self.current_system.name == 'pilatus':
+            self.valid_prog_environs = ['cpeIntel']
+        else:
+            self.valid_prog_environs = ['builtin']
```
Comment (vkarak) [inline cscs-checks/apps/namd/namd_check.py:56]: I would put all the valid prog. environments at the beginning and, here, I would remove those that are not valid. This makes the test, I think, more readable.

File: cscs-checks/libraries/gridtools/gridtools_tests.py [hunk 2]
```diff
@@ -29,50 +35,58 @@ def __init__(self):
...
+    @rfm.run_before('sanity')
+    def set_sanity_patterns(self):
         self.sanity_patterns = sn.assert_found(r'perftest',
                                                self.stdout)
-        self.tags = {'scs', 'benchmark'}
-        self.maintainers = ['CB']
 
 
 @rfm.simple_test
 class GridToolsCPUBuildCheck(GridToolsBuildCheck):
```
Comment (vkarak) [inline cscs-checks/libraries/gridtools/gridtools_tests.py:46]: I would rather have a single build test parametrised over `cpu` and `gpu`.

Comment (ChristopherBignamini) [reply cscs-checks/libraries/gridtools/gridtools_tests.py]: Done

File: cscs-checks/apps/namd/namd_check.py (hunk 1 above)
Comment (ChristopherBignamini) [reply cscs-checks/apps/namd/namd_check.py]: Done

File: cscs-checks/apps/namd/namd_check.py [hunk 3]
```diff
@@ -10,23 +10,29 @@
...
+        }
+    }
 
-        # Reset sources dir relative to the SCS apps prefix
-        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
-                                       'NAMD', 'prod')
-        self.executable = 'namd2'
-        self.use_multithreading = True
-        self.num_tasks_per_core = 2
+    def set_description(self, arch):
```
[the remark this answers is not in the document]
Comment (ChristopherBignamini) [reply cscs-checks/apps/namd/namd_check.py]: Done

**Comment by ChristopherBignamini** [comment]
@jenkins-cscs retry dom

File: cscs-checks/libraries/gridtools/gridtools_tests.py [hunk 4]
```diff
@@ -80,105 +84,122 @@ def __init__(self):
...
-                        'wall_time': (1.09, None, 0.05, 's')
-                    },
-                    'dom:mc': {
-                        'wall_time': (11.0, None, 0.05, 's')
-                    },
-                    'dom:gpu': {
-                        'wall_time': (1.09, None, 0.05, 's')
-                    }
+    variant = parameter(['horizontal_diffusion/cpu_kfirst_double',
+                        'horizontal_diffusion/cpu_ifirst_double'])
```
Comment (teojgo) [inline cscs-checks/libraries/gridtools/gridtools_tests.py:88]:
```suggestion
                         'horizontal_diffusion/cpu_ifirst_double'])
```

**Review by teojgo — APPROVED** [review]

**Review by vkarak — CHANGES_REQUESTED** [review]
@ChristopherBignamini I've done some style fixes, but the GridTools checks are failing. Could you have a look?

**Comment by ChristopherBignamini** [comment]
Thank you @vkarak , the GridTools test seems to be ok now.

File: cscs-checks/libraries/gridtools/gridtools_tests.py [hunk 5]
```diff
@@ -72,13 +71,13 @@ class GridToolsRunCheck(rfm.RunOnlyRegressionTest):
     def validate_run(self):
         return sn.assert_found(r'PASSED', self.stdout)
 
+    @deferrable
     @run_before('performance')
```
Comment (vkarak) [inline cscs-checks/libraries/gridtools/gridtools_tests.py]: I am changing that to `@performance_function`

File: cscs-checks/libraries/gridtools/gridtools_tests.py [hunk 6]
```diff
@@ -72,13 +71,13 @@ class GridToolsRunCheck(rfm.RunOnlyRegressionTest):
...
     def setup_perf_vars(self):
-        literal_eval = sn.sanity_function(ast.literal_eval)
         self.perf_patterns = {
-            'wall_time': sn.avg(literal_eval(
+            'wall_time': sn.avg(
                 sn.extractsingle(r'"series" : \[(?P<wall_times>.+)\]',
-                                 self.stdout, 'wall_times')))
+                                 self.stdout, 'wall_times'))
```
Comment (vkarak) [inline cscs-checks/libraries/gridtools/gridtools_tests.py:80]: I'm not sure that these changes are correct.

**Review by vkarak — CHANGES_REQUESTED** [review]
I will push the fixes in a moment.

**Review by vkarak — APPROVED** [review]

**Pull request merged**